src/BugsnagUnity/Native/Cocoa/NativePayloadClassWrapper.cs

```diff
@@ -32,7 +36,44 @@ internal void SetNativeBool(string name, bool? value)
             NativeCode.bugsnag_setBoolValue(NativePointer, name, stringValue);
         }
 
-        internal DateTime? GetNativeTimestamp(string key)
+        internal double GetNativeDouble(string key)
+        {
+            return NativeCode.bugsnag_getDoubleValue(NativePointer, key);
+        }
+
+        internal void SetNativeDouble(string key, double value)
+        {
+            NativeCode.bugsnag_setDoubleValue(NativePointer,key,value);
+        }
+
+        internal long? GetNativeLong(string key)
+        {
+            long value = NativeCode.bugsnag_getLongValue(NativePointer, key);
+            if (value < 0)
+            {
+                return null;
+            }
+            else
+            {
+                return value;
+            }
+        }
+
+        internal void SetNativeLong(string key, long? value)
+        {
+
+            if (value == null)
+            {
+                NativeCode.bugsnag_setLongValue(NativePointer, key, -1);
+            }
+            else
+            {
+                NativeCode.bugsnag_setLongValue(NativePointer,key, (long)value);
+            }
+            
+        }
```

Comment:
The use of `-1` to indicate `null` is problematic because `-1` could be a valid value.

Reply:
any suggestions on how to handle this? I can't pass null.

Reply:
Fixed 👍